var/spack/repos/builtin/packages/meme/package.py

```diff
@@ -35,9 +35,20 @@ class Meme(AutotoolsPackage):
 
     version('4.11.4', '371f513f82fa0888205748e333003897')
 
+    variant('mpi', default=False, description='Enable MPI support')
+
     depends_on('zlib', type=('link'))
     depends_on('libxml2', type=('link'))
     depends_on('libxslt', type=('link'))
     depends_on('libgcrypt', type=('link'))
     depends_on('perl', type=('build', 'run'))
     depends_on('python@2.7:', type=('build', 'run'))
+    # openmpi support
+    depends_on('openmpi@2:', when='+mpi', type=('build', 'run'))
```

Comment:
When I remove link for zlib libxml2 libxslt libgcrypt and openmpi we're back at using LD_LIBRARY_PATH

```
user@node103[~/repos/spack]$ ldd $(which meme)
        linux-vdso.so.1 =>  (0x00007fff53fff000)
        libexslt.so.0 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/libxslt-1.1.29-44723n7a6zoh3hw3h2epwsvwetnaxkpt/lib/libexslt.so.0 (0x00007fe7e6172000)
        libxslt.so.1 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/libxslt-1.1.29-44723n7a6zoh3hw3h2epwsvwetnaxkpt/lib/libxslt.so.1 (0x00007fe7e5f34000)
        libgcrypt.so.20 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/libgcrypt-1.6.2-icbvp2oy6amw4epfjbm442rwyejjodcw/lib/libgcrypt.so.20 (0x00007fe7e5c6e000)
        libgpg-error.so.0 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/libgpg-error-1.21-jzkx2yr3ctlxks525xo6yo2xap6ievnj/lib/libgpg-error.so.0 (0x00007fe7e5a5c000)
        libxml2.so.2 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/libxml2-2.9.4-szkxoqturs2fzrvsrzz4iolbnl33qjdf/lib/libxml2.so.2 (0x00007fe7e56fe000)
        liblzma.so.5 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/xz-5.2.3-7ah2pytuetgk63x5scnbdoma63lvijvz/lib/liblzma.so.5 (0x00007fe7e54d9000)
        librt.so.1 => /lib64/librt.so.1 (0x0000003dba400000)
        libdl.so.2 => /lib64/libdl.so.2 (0x0000003db9000000)
        libmpi.so.20 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/openmpi-2.1.1-2jvf43qz5oprklkls4jvxeq5nlrgeu2a/lib/libmpi.so.20 (0x00007fe7e50cd000)
        libz.so.1 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/zlib-1.2.11-erzei6xutxne2tjssbmelgutsnhz6x6s/lib/libz.so.1 (0x00007fe7e4eaf000)
        libm.so.6 => /lib64/libm.so.6 (0x0000003db9c00000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x0000003db9800000)
        libc.so.6 => /lib64/libc.so.6 (0x0000003db9400000)
        /lib64/ld-linux-x86-64.so.2 (0x0000003db8c00000)
        libgpfs.so => /usr/lib64/libgpfs.so (0x00007fe7e4da2000)
        libopen-rte.so.20 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/openmpi-2.1.1-2jvf43qz5oprklkls4jvxeq5nlrgeu2a/lib/libopen-rte.so.20 (0x00007fe7e4a99000)
        libopen-pal.so.20 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/openmpi-2.1.1-2jvf43qz5oprklkls4jvxeq5nlrgeu2a/lib/libopen-pal.so.20 (0x00007fe7e473a000)
        librdmacm.so.1 => /usr/lib64/librdmacm.so.1 (0x0000003210a00000)
        libibverbs.so.1 => /usr/lib64/libibverbs.so.1 (0x0000003210200000)
        libutil.so.1 => /lib64/libutil.so.1 (0x0000003dc9e00000)
        libhwloc.so.5 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/hwloc-1.11.6-7oifsdjotrd6vdidxdfgdr5nbalbfxik/lib/libhwloc.so.5 (0x00007fe7e44fb000)
        libnuma.so.1 => /usr/lib64/libnuma.so.1 (0x0000003dd0e00000)
        libudev.so.0 => /lib64/libudev.so.0 (0x0000003dc0400000)
        libpciaccess.so.0 => /pbtech_mounts/homes027/user/repos/spack/opt/spack/linux-rhel6-x86_64/gcc-4.4.6/libpciaccess-0.13.4-7tleclpx332774jbjbf5a5rgjqpryres/lib/libpciaccess.so.0 (0x00007fe7e42f2000)
        libnl.so.1 => /lib64/libnl.so.1 (0x0000003dbf800000)
```